### PR TITLE
Moving php-pear to core & adding php-pecl-zip

### PR DESCRIPTION
**Pull request by osamaesmailmsft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
PHP is more useful with its package manager (PECL) that's built by php-pear, so it moves from extended to core. We also had a request to added the extension php-pecl-zip to core.

###### Change Log  

- Moves php-pear from SPECS-EXTENDED to SPECS
- Adds php-pecl-zip to SPECS

###### Does this affect the toolchain?  

NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Buddy Build (ARM64: 257439, AMD64: 257438)

### CODE REVIEW

File: SPECS/php-pecl-zip/php-pecl-zip.spec [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+%global with_zts  0%{?__ztsphp:1}
+%global ini_name  40-%{pecl_name}.ini
+
+Summary:      A ZIP archive management extension
+Summary(fr):  Une extension de gestion des ZIP
+Name:         php-pecl-zip
+Version:      1.21.1
+Release:      3%{?dist}
+License:      PHP
+URL:          https://pecl.php.net/package/zip
```
Comment (PawelWMS) [inline SPECS/php-pecl-zip/php-pecl-zip.spec:23]: I think we're missing the `Distribution` and `Vendor` tags.

File: SPECS/php-pecl-zip/php-pecl-zip.spec [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+BuildRequires:  php-devel
+BuildRequires:  php-pear
+BuildRequires:  pkg-config
+BuildRequires:  zlib-devel
+BuildRequires:  pkgconfig(libzip) >= 1.0.0
+Requires:       php(api) = 20210902-%{__isa_bits}
+Requires:       php(zend-abi) = 20210902-%{__isa_bits}
+# Supposed to use these macros from SPECS/php/macros.php
+#Requires:     php(zend-abi) = %{php_zend_api}
+#Requires:     php(api) = %{php_core_api}
```
Comment (PawelWMS) [inline SPECS/php-pecl-zip/php-pecl-zip.spec:34]:
```suggestion

# The 'php_*_api' macros are dynamically generated during php.spec build time,
# so these macros don't work with the way Mariner uses macros during spec parsing.
# Once that issue is resolved, we should bring back the original values.
#Requires:     php(zend-abi) = %{php_zend_api}
#Requires:     php(api) = %{php_core_api}
Requires:       php(api) = 20210902-%{__isa_bits}
Requires:       php(zend-abi) = 20210902-%{__isa_bits}

```

File: SPECS/php-pear/php-pear.spec [hunk 3]
```diff
@@ -70,19 +60,21 @@ Requires:       php-posix
...
 Provides:       php-pear(Structures_Graph) = %{structver}
 Provides:       php-pear(XML_Util) = %{xmlutil}
 Provides:       php-pear(PEAR_Manpages) = %{manpages}
-
 Provides:       php-composer(pear/console_getopt) = %{getoptver}
 Provides:       php-composer(pear/archive_tar) = %{arctarver}
 Provides:       php-composer(pear/pear-core-minimal) = %{version}
 Provides:       php-composer(pear/structures_graph) = %{structver}
 Provides:       php-composer(pear/xml_util) = %{xmlutil}
+BuildArch:      noarch
```
Comment (PawelWMS) [inline SPECS/php-pear/php-pear.spec:74]: Could you move this between the sources and build dependencies? Linter moving it here is one of the quirks we still have to deal with.

File: SPECS/php-pear/php-pear.spec [hunk 4]
```diff
@@ -70,19 +60,21 @@ Requires:       php-posix
...
+%if %{with_check}
+BuildRequires:  %{_bindir}/phpunit
```
Comment (PawelWMS) [inline SPECS/php-pear/php-pear.spec:76]: From the buddy builds it looks like we're missing this one, unfortunately.

File: SPECS/php-doctrine-instantiator/php-doctrine-instantiator.spec [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+
+%files
+%license LICENSE
+%doc *.md composer.json
+%dir %{_datadir}/php/Doctrine
+     %{_datadir}/php/Doctrine/Instantiator%{major}
+
+
+%changelog
```
Comment (neha170) [inline SPECS/php-doctrine-instantiator/php-doctrine-instantiator.spec:115]: Please update changelog, and bump up the release number

File: SPECS/php-pecl-zip/php-pecl-zip.spec [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+# http://creativecommons.org/licenses/by-sa/4.0/
+#
+# Please, preserve the changelog entries
+#
+
+%global pecl_name zip
+%global with_zts  0%{?__ztsphp:1}
+%global ini_name  40-%{pecl_name}.ini
+Summary:        A ZIP archive management extension
+Summary(fr):    Une extension de gestion des ZIP
```
Comment (neha170) [inline SPECS/php-pecl-zip/php-pecl-zip.spec:14]: Please remove this line

File: SPECS/php-pecl-zip/php-pecl-zip.spec [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+Provides:       php-pecl(%{pecl_name})%{?_isa} = %{version}
+Provides:       php-%{pecl_name} = %{version}-%{release}
+Provides:       php-%{pecl_name}%{?_isa} = %{version}-%{release}
+# we don't want -z defs linker flag
+%undefine _strict_symbol_defs_build
+
+%description
+Zip is an extension to create and read zip files.
+
+%description -l fr
```
Comment (neha170) [inline SPECS/php-pecl-zip/php-pecl-zip.spec:45]: Please remove this line

File: SPECS/php-phar-io-manifest2/php-phar-io-manifest2.spec [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+BuildRequires:  php-phar
+BuildRequires: (php-composer(phar-io/version) >= 3.0.1 with php-composer(phar-io/version) <  4)
+BuildRequires:  php-filter
+BuildRequires:  php-libxml
+BuildRequires:  php-pcre
+BuildRequires:  php-spl
+BuildRequires:  php-xmlwriter
+BuildRequires:  php-fedora-autoloader-devel >= 1.0.0
+%if %{with_check}
+%if 0%{?fedora} >= 32
```
Comment (neha170) [inline SPECS/php-phar-io-manifest2/php-phar-io-manifest2.spec:34]: Please remove all references to fedora from the spec. Kindly re-evaluate the if-else statements wherever applicable, thanks

File: SPECS/php-phar-io-manifest2/php-phar-io-manifest2.spec [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+%endif
+
+
+%files
+%license LICENSE
+%doc README.md composer.json
+%{php_home}/PharIo/Manifest2
+
+
+%changelog
```
Comment (neha170) [inline SPECS/php-phar-io-manifest2/php-phar-io-manifest2.spec:117]: Kindly add an entry in the changelog, and update the release

File: SPECS/php-sebastian-diff4/php-sebastian-diff4.spec [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+%{_bindir}/phpab --template fedora --output src/autoload.php src
+
+
+%install
+mkdir -p   %{buildroot}%{php_home}/SebastianBergmann
+cp -pr src %{buildroot}%{php_home}/SebastianBergmann/Diff4
+
+
+%check
+%if %{with_check}
```
Comment (neha170) [inline SPECS/php-sebastian-diff4/php-sebastian-diff4.spec:71]: Please remove the if-else-end statement, this block will run only when tests are enabled

File: SPECS/php-sebastian-diff4/php-sebastian-diff4.spec [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+
+%files
+%license LICENSE
+%doc composer.json
+%doc *.md
+%dir %{php_home}/SebastianBergmann
+     %{php_home}/SebastianBergmann/Diff4
+
+
+%changelog
```
Comment (neha170) [inline SPECS/php-sebastian-diff4/php-sebastian-diff4.spec:106]: Update changelog and bump up release

File: SPECS/php-doctrine-instantiator/php-doctrine-instantiator.spec [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+
+Name:           php-doctrine-instantiator
+Version:        1.4.1
+Release:        2%{?dist}
+Summary:        Instantiate objects in PHP without invoking their constructors
+
+License:        MIT
+URL:            https://github.com/doctrine/instantiator
+Source0:        https://github.com/doctrine/instantiator/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
+#Source1:        makesrc.sh
```
Comment (neha170) [inline SPECS/php-doctrine-instantiator/php-doctrine-instantiator.spec:27]: Can remove this line

File: SPECS/php-phar-io-manifest2/php-phar-io-manifest2.spec [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+
+Name:           php-phar-io-manifest2
+Version:        2.0.3
+Release:        4%{?dist}
+Summary:        Component for reading phar.io manifest information
+
+License:        BSD
+URL:            https://github.com/phar-io/manifest
+Source0:        https://github.com/phar-io/manifest/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
+#Source1:        makesrc.sh
```
Comment (neha170) [inline SPECS/php-phar-io-manifest2/php-phar-io-manifest2.spec:20]: Please remove this line

File: SPECS/php-sebastian-diff4/php-sebastian-diff4.spec [hunk 14]
```diff
@@ -0,0 +1,200 @@
+# remirepo/fedora spec file for php-sebastian-diff4
+#
+# Copyright (c) 2013-2020 Remi Collet
+# License: CC-BY-SA
+# http://creativecommons.org/licenses/by-sa/4.0/
+#
+# Please, preserve the changelog entries
+#
+
+%bcond_without       tests
```
Comment (neha170) [inline SPECS/php-sebastian-diff4/php-sebastian-diff4.spec:10]: Please remove this condition, as our build tags take care of running check section. Thanks

File: SPECS/php-sebastian-diff4/php-sebastian-diff4.spec [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+
+Name:           php-sebastian-diff4
+Version:        4.0.4
+Release:        5%{?dist}
+Summary:        Diff implementation, version 4
+
+License:        BSD
+URL:            https://github.com/sebastianbergmann/diff
+Source0:        https://github.com/sebastianbergmann/diff/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
+#Source1:        makesrc.sh
```
Comment (neha170) [inline SPECS/php-sebastian-diff4/php-sebastian-diff4.spec:23]: Can remove this line

File: SPECS/php-sebastian-exporter4/php-sebastian-exporter4.spec [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+
+Name:           php-sebastian-exporter4
+Version:        4.0.5
+Release:        1%{?dist}
+Summary:        Export PHP variables for visualization, version 4
+
+License:        BSD
+URL:            https://github.com/sebastianbergmann/exporter
+Source0:        https://github.com/sebastianbergmann/exporter/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
+#Source1:        makesrc.sh
```
Comment (neha170) [inline SPECS/php-sebastian-exporter4/php-sebastian-exporter4.spec:21]: Remove this line

File: SPECS/php-sebastian-exporter4/php-sebastian-exporter4.spec [hunk 17]
```diff
@@ -0,0 +1,195 @@
...
+require_once '%{php_home}/SebastianBergmann/RecursionContext4/autoload.php';
+EOF
+
+
+%install
+mkdir -p   %{buildroot}%{php_home}/SebastianBergmann
+cp -pr src %{buildroot}%{php_home}/SebastianBergmann/Exporter4
+
+
+%if %{with_check}
```
Comment (neha170) [inline SPECS/php-sebastian-exporter4/php-sebastian-exporter4.spec:73]: Remove this if statement

File: SPECS/php-sebastian-object-enumerator4/php-sebastian-object-enumerator4.spec [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+
+Name:           php-sebastian-object-enumerator4
+Version:        4.0.4
+Release:        2
+Summary:        Traverses array and object to enumerate all referenced objects, version 4
+
+License:        BSD
+URL:            https://github.com/sebastianbergmann/object-enumerator
+Source0:        https://github.com/sebastianbergmann/object-enumerator/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
+#Source1:        makesrc.sh
```
Comment (neha170) [inline SPECS/php-sebastian-object-enumerator4/php-sebastian-object-enumerator4.spec:21]: Remove this line

File: SPECS/php-sebastian-object-enumerator4/php-sebastian-object-enumerator4.spec [hunk 19]
```diff
@@ -0,0 +1,182 @@
...
+
+
+%files
+%license LICENSE
+%doc README.md composer.json
+%{php_home}/SebastianBergmann/ObjectEnumerator4
+
+
+%changelog
+* Fri Jul 22 2022 Fedora Release Engineering <releng@fedoraproject.org> - 4.0.4-1.4
```
Comment (neha170) [inline SPECS/php-sebastian-object-enumerator4/php-sebastian-object-enumerator4.spec:104]: Please add a changelog entry and bump up the release

File: SPECS/phpunit9/phpunit9.spec [hunk 20]
```diff
@@ -0,0 +1,1533 @@
+# For compatibility with SCL
+%undefine __brp_mangle_shebangs
+
+# Namespace
+%global ns_vendor    PHPUnit9
+%global php_home     %{_datadir}/php
+
+Name:           phpunit9
+Version:        9.5.26
+Release:        1%{?dist}
```
Comment (neha170) [inline SPECS/phpunit9/phpunit9.spec:10]: Release should be 2

**Review by neha170 — COMMENTED** [review]
Thanks for making the PR. Few comments,
Please remove files
- SPECS/php-pecl-zip/php-pecl-zip-1.21.1.tgz
- SPECS/php-phar-io-manifest2/php-phar-io-manifest2-2.0.3.tar.gz
- SPECS/php-sebastian-exporter4/php-sebastian-exporter4-4.0.5.tar.gz
- SPECS/phpunit9/phpunit9-9.5.26.tar.gz

Please revert the change in toolkit/resources/manifests/package/*.txt files and rebuild the packages with the latest toolchain, thanks

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**